### PR TITLE
Replace change server input with a dropdown menu

### PR DESCRIPTION
**Pull request by saruwman**
[Replacing the change server input field with a select dropdown containing: Default server (Ferdi), Franz Server, and a custom option allowing the user to input a custom server](https://github.com/getferdi/ferdi/issues/706#event-3297586151)

### CODE REVIEW

**Comment by saruwman** [comment]
image
image
image

**Comment by saruwman** [comment]
ferdichangeserver

File: src/i18n/messages/src/components/auth/ChangeServer.json [hunk 1]
```diff
@@ -2,39 +2,78 @@
   {
     "id": "changeserver.headline",
     "defaultMessage": "!!!Change server",
-    "file": "src/components/auth/ChangeServer.js",
+    "file": "src\\components\\auth\\ChangeServer.js",
```
Comment (kytwb) [inline src/i18n/messages/src/components/auth/ChangeServer.json:5]: I am not sure why some of the `/` got changed into `\\` here. 
@sampathBlam @mahadevans87 is this something you encountered while developing on Windows?

Comment (sampathBlam) [reply src/i18n/messages/src/components/auth/ChangeServer.json:5]: Yes windows does that. And we probably shouldn't checkin that piece of change.

Comment (mahadevans87) [reply src/i18n/messages/src/components/auth/ChangeServer.json:5]: Can we fix that by disabling that in eslint?

Comment (sampathBlam) [reply src/i18n/messages/src/components/auth/ChangeServer.json:5]: I don't remember if it is because of lint or just running ferdi source in a windows pc itself does that.

Comment (mahadevans87) [reply src/i18n/messages/src/components/auth/ChangeServer.json:5]: Ok. I can confirm that the changes occur when you run `npm run dev` which basically calls `cross-env NODE_ENV=development gulp dev`

Comment (mahadevans87) [reply src/i18n/messages/src/components/auth/ChangeServer.json:5]: Ok Some more information - This happens in the gulp.babel file especially when the scripts function is ran - 

`export function scripts() {
  return gulp
    .src(paths.scripts.src, { since: gulp.lastRun(scripts) })
    .pipe(
      babel({
        comments: false
      })
    )
    .pipe(gulpIf(process.env.NODE_ENV !== "development", uglify())) // Only uglify in production to speed up dev builds
    .pipe(gulp.dest(paths.scripts.dest));
}`

Comment (mahadevans87) [reply src/i18n/messages/src/components/auth/ChangeServer.json:5]: In .babelrc file, 
```
"@babel/syntax-dynamic-import",
    ["react-intl", {
      "messagesDir": "./src/i18n/messages/",
      "enforceDescriptions": false,
      "extractSourceLocation": true
    }]
```

If you set `extractSourceLocation` to `false' react-intl won't output the file, start, line and column properties in each messages.json file. However, if we want to retain extractSourceLocation: true, Windows developers have to live with this I guess.

Comment (kytwb) [reply src/i18n/messages/src/components/auth/ChangeServer.json]: Thank you for investigating guys 🙏 If it doesn't introduce regressions, I'm for setting `extractSourceLocation` to false. I don't think we use the location details and it would help enhance the readability of those files.

File: src/components/ui/Input.js [hunk 2]
```diff
@@ -72,7 +72,7 @@ export default @observer class Input extends Component {
     if (type === 'password' && this.state.showPassword) {
       type = 'text';
     }
-
+
```
Comment (kytwb) [inline src/components/ui/Input.js:75]: @saruwman Minor note for next time; if you go to https://github.com/getferdi/ferdi/pull/717/files, you can see a nice overview of the file changes you are submitting in your pull request:

Screen Shot 2020-05-17 at 8 20 51 AM

It can help spot unnecessary changes before submitting for review 🙏
